core/src/main/java/org/kiwix/kiwixmobile/core/webserver/ZimHostActivity.java

```diff
@@ -305,4 +296,27 @@ private void launchTetheringSettingsScreen() {
       Toast.LENGTH_SHORT)
       .show();
   }
+
+  private void selectPreviouslyHostedBooks(@Nullable List<BooksOnDiskListItem> books) {
+    selectedBooksId.addAll(sharedPreferenceUtil.getHostedBooks());
+    if (books != null && !books.isEmpty()) {
+      if (selectedBooksId.isEmpty()) {
+        // Select all books if no book ids are stored
+        for (BooksOnDiskListItem book : books) {
+          if (book instanceof BooksOnDiskListItem.BookOnDisk) {
+            selectedBooksId.add(((BooksOnDiskListItem.BookOnDisk) book).getBook().getId());
+            book.setSelected(true);
+          }
+        }
+      } else {
+        for (BooksOnDiskListItem book : books) {
+          if (book instanceof BooksOnDiskListItem.BookOnDisk) {
```

Comment:
the instanceOf check is not needed if you have a longSet which uses the listItem Id instead of the bookId